src/internal/types/themeData.ts

```diff
@@ -0,0 +1,15 @@
+export type ThemeSaveState = {
+  history: any; // json object
+  index: number;
+  hash: string;
+};
+
+export type ThemeHistories = {
+  histories: History[];
+  index?: number;
+};
+
+export type History<D = any> = {
```

Comment:
This shouldn't need to be generic. Themes have a specific format, which is just an array of strings.